kmp_cpu.cpp

```diff
@@ -3,30 +3,66 @@
 static void get_fail(const char *pattern, int pattern_length, int *fail) {
```

Comment:
Change the type of pattern_length to int16_t, and fail to int16_t* for consistency.

Reply:
changed